spec/immutable-data-structures.html

```diff
@@ -6,26 +6,521 @@ <h1>Immutable Data Structures</h1>
+          1. Let _list_ be ? CreateListFromTuple(_T_)
+          1. Let _len_ be the number of elements in _list_.
+          1. Let _relativeStart_ be ? ToInteger(_start_).
+          1. If _relativeStart_ &lt; 0, let _actualStart_ be max((_len_ + _relativeStart_), 0); else let _actualStart_ be min(_relativeStart_, _len_).
+          1. If the number of actual arguments is 0, then
+            1. Let _insertCount_ be 0.
+            1. Let _actualDeleteCount_ be 0.
+          1. Else if the number of actual arguments is 1, then
+            1. Let _insertCount_ be 0.
+            1. Let _actualDeleteCount_ be _len_ - _actualStart_.
+          1. Else,
+            1. Let _insertCount_ be the number of actual arguments minus 2.
+            1. Let _dc_ be ? ToInteger(_deleteCount_).
+            1. Let _actualDeleteCount_ be min(max(_dc_, 0), _len_ - _actualStart_).
+          1. If _len_ + _insertCount_ - _actualDeleteCount_ &gt; 2<sup>53</sup> - 1, throw a *TypeError* exception.
+          1. Let _k_ be 0.
+          1. Let _items_ be a List whose elements are, in left to right order, the portion of the actual argument list starting with the third argument. The list is empty if fewer than three arguments were passed.
+          1. Let _itemCount_ be the number of elements in _items_.
+          1. Let _newList_ be a new empty List.
+          1. Repeat, while _k_ &lt; _actualStart_,
+            1. Let _E_ be _list_[_k_].
+            1. Append _E_ to the end of list _newList_.
+            1. Set _k_ to _k_ + 1.
+          1. Let _itemK_ be 0.
+          1. Repeat, while _itemK_ &lt; _itemCount_.
+            1. Let _E_ be _items_[_itemK_].
+            1. Append _E_ to the end of _newList_.
+            1. Set _itemK_ to _itemK_ + 1.
+          1. Set _k_ to _actualStart_ + _actualDeleteCount_.
+          1. Repeat, while _k_ &lt; _len_,
+            1. Let _E_ be _list_[_k_].
+            1. Append _E_ to the end of _newList_.
+            1. Set _k_ to _k_ + 1.
+          1. Return ? CreateTupleFromList(_newList_).
+        </emu-alg>
+      </emu-clause>
+      <emu-clause id="sec-tuple.prototype.concat">
+        <h1>Tuple.prototype.concat ( ..._args_ )</h1>
+        <p>When the *concat* method is called with zero or more arguments, it returns a tuple containing the elements of the tuple followed by the elements of each argument in order.</p>
+        <p>The following steps are taken:</p>
+        <emu-alg>
+          1. Let _T_ be ? thisTupleValue(*this* value).
+          1. Let _list_ be a new empty List.
+          1. Let _n_ be 0.
+          1. Let _items_ be a List whose first element is _T_ and whose subsequent element are, in left to right order, the arguments that were passed to this function invocation.
+          1. Repeat, while _items_ is not empty,
+            1. Remove the first element from _items_ and let _E_ be the value of the element.
+            1. Let _spreadable_ be ? IsConcatSpreadable(_E_).

```

Comment:
I guess we should make sure that Tuples are concat spreadable (either by updating that algorithm or giving them the symbol; I prefer the former).

Reply:
It's a protocol; it's fine to have default behavior here, but I think the symbol needs to exist and have its value respected.

Reply:
I'm not sure what you're trying to say here. Nothing in the JS spec has that symbol defined. I'm not saying that we would stop querying the symbol, just that it would have this default behavior built into the IsConcatSpreadable algorithm, like Arrays do.